src/processors/other/Tremolo.cpp

```diff
@@ -121,26 +123,46 @@ void Tremolo::processAudio (AudioBuffer<float>& buffer)
     phaseSmooth.setTargetValue (*rateParam * MathConstants<float>::pi / fs);
     waveSmooth.setTargetValue (*waveParam);
 
-    // fill wave buffer (-1, 1)
-    dsp::AudioBlock<float> waveBlock { waveBuffer };
-    dsp::ProcessContextReplacing<float> waveCtx { waveBlock };
-    fillWaveBuffer (waveBlock.getChannelPointer (0), numSamples, phase);
-
-    // shrink range to (0, 1)
-    waveBlock *= 0.5f;
-    waveBlock += 0.5f;
-
-    // apply depth parameter
-    auto depthVal = std::pow (depthParam->load(), 0.33f);
-    depthGainSmooth.setTargetValue (depthVal);
-    waveBlock.multiplyBy (depthGainSmooth);
-    depthAddSmooth.setTargetValue (1.0f - depthVal);
-    addSmoothed (waveBlock, depthAddSmooth);
-    filter.process (waveCtx);
+    if (inputsConnected.contains (1))
+    {
+        waveBuffer.copyFrom (0, 0, getInputBuffer (1), 0, 0, numSamples);
+        for (int ch = 0; ch < buffer.getNumChannels(); ++ch)
+        {
+            auto* x = getInputBuffer (0).getWritePointer (0);
+            FloatVectorOperations::multiply (x, x, waveBuffer.getReadPointer (0), numSamples);
+        }
+    }
+    else
+    {
+        // fill wave buffer (-1, 1)
+        dsp::AudioBlock<float> waveBlock { waveBuffer };
+        dsp::ProcessContextReplacing<float> waveCtx { waveBlock };
+        fillWaveBuffer (waveBlock.getChannelPointer (0), numSamples, phase);
+
+        // shrink range to (0, 1)
+        waveBlock *= 0.5f;
+        waveBlock += 0.5f;
+
+        // apply depth parameter
+        auto depthVal = std::pow (depthParam->load(), 0.33f);
+        depthGainSmooth.setTargetValue (depthVal);
+        waveBlock.multiplyBy (depthGainSmooth);
+        depthAddSmooth.setTargetValue (1.0f - depthVal);
+        addSmoothed (waveBlock, depthAddSmooth);
+        filter.process (waveCtx);
+
+        for (int ch = 0; ch < buffer.getNumChannels(); ++ch)
+        {
+            auto* x = getInputBuffer (0).getWritePointer (0);
+            FloatVectorOperations::multiply (x, x, waveBuffer.getReadPointer (0), numSamples);
+        }
+    }
 
-    for (int ch = 0; ch < buffer.getNumChannels(); ++ch)
+    if (! inputsConnected.contains (0)) //If normal input is connected no sounds goes through, only modulation signal
     {
-        auto* x = buffer.getWritePointer (ch);
-        FloatVectorOperations::multiply (x, x, waveBuffer.getReadPointer (0), numSamples);
+        buffer.clear();
     }
+
+    outputBuffers.getReference (0) = &getInputBuffer (0);
```

Comment:
Yeah, I found that the modulation signal is added to the `buffer` parameter in a situation where I had 2 tremolos.

In the attached picture, using `buffer` instead of `getInputBuffer()`makes the modulation signal be sent to the output instead of just silence/The Input like it should be.

I figured the two signals were being added to the input buffer somewhere in the chain and using the individual `getInputBuffer()`'s seemed like a way of fixing this. Do you think there is a better solution?

![Screen Shot 2022-09-22 at 11 14 58 AM](https://user-images.githubusercontent.com/43660096/191821414-0af9df01-d630-486e-aa7e-db2c033f4f7b.png)

Reply:
Oh yeah... I should definitely document this somewhere (probably in BaseProcessor.h?), since I forgot about it since the last time I worked with it.

Basically for modules with multiple inputs or outputs, we should ignore the buffer that's passed in through the function argument, and just use `getInputBuffer()` or `getOutputBuffer()` for everything. Sorry about that, I'll make sure to get it documented.

Reply:
All good! I was definitely confused by that for a second but luckily the other modules do a good job of using the other buffers.